test/serial-tests/test_window_suspend.c

```diff
@@ -0,0 +1,186 @@
+/* Test libnxz behavior when windows are suspended due to a DLPAR operation.
+
+   Test outline:
+   - Consume N-1 credits by calling deflateInit from N-1 child processes
+   - Force a DLPAR core remove operation to remove 3 credits from the system
+   - The last 2 processes created have suspended windows and the system is
+     oversubscribed by 2 credits.
+   - Release 3 credits by finishing processes with active windows
+   - Now the processes with suspended windows should be able to get a new one
+     and finish execution successfully.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/fcntl.h>
+#include <sys/mman.h>
+#include <endian.h>
+#include <bits/endian.h>
+#include <sys/ioctl.h>
+#include <sys/wait.h>
+#include <assert.h>
+#include <errno.h>
+#include <signal.h>
+
+#include "test.h"
+#include "test_utils.h"
+#include "credit_utils.h"
+
+#define CPU_ARGS "-c cpu -w 5 -d 3"
+
+Bytef *src, *compr;
+const unsigned int src_len = 32000; 	/* Need something larger than
+					   cache_threshold to avoid deflate
+					   falling back to software by
+					   default. */
+const unsigned int compr_len = 64000;
+
+/* TODO: calling system with super-user privileges is not recommended, implement
+   our own version using fork, exec and wait instead. */
+int run(const char *command)
+{
+	return system(command);
+}
+
+/* For some reason the number of credits only changes when we alter the number
+   of virtual processors, even though on shared LPAR they are tied to the number
+   of processing units (aka Entitled Capactity). Remember credits=proc_units*20,
+   and drmgr sees 0.01 proc_units as 1 ent_capacity, so reducing ent_capacity by
+   5 means removing 0.05 units, and thus 1 credit. */
+#define PROC_UNITS "15" // -3 credits
+#define CPU_COUNT "1"
+
+int reduce_credits(void)
+{
+	run("drmgr " CPU_ARGS " -r -p ent_capacity -q " PROC_UNITS);
+	run("drmgr " CPU_ARGS " -r -q " CPU_COUNT);
+
+	return 0;
+}
+
+void restore_credits(void)
+{
+	(void) run("drmgr " CPU_ARGS " -a -p ent_capacity -q " PROC_UNITS);
+	(void) run("drmgr " CPU_ARGS " -a -q " CPU_COUNT);
+}
+
+/* Work for the child */
+void child_do(z_stream *stream) {
+	int rc;
+
+	stream->next_in = src;
+	stream->avail_in = src_len;
+	stream->next_out = compr;
+	stream->avail_out = compr_len;
+
+	rc = deflate(stream, Z_FINISH);
+	if (rc != Z_STREAM_END) {
+		fprintf(stderr, "deflate failed! rc = %s\n", zret2str(rc));
+		exit(1);
+	}
+}
+
+int main(int argc, char** argv)
+{
+	int total_credits, used_credits;
+	int status;
+
+	/* The credit system is only used by PowerVM. So skip otherwise. */
+	if (!is_powervm())
+		return TEST_SKIP;
+
+	/* Need to be root to run drmgr */
+	if (geteuid() != 0) {
+		fprintf(stderr, "This test needs to be run as root.\n");
+		return TEST_SKIP;
```

Comment:
This is not going to fly....
In most cases this test will be skipped because nobody runs `make check` as root and nobody should.
I understand the problem, but we might need to look at a simple container-like structure to guarantee this test is properly executed.
Considering this won't fail, LGTM for now.